src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    Node<T> node(int index) {
+        Node<T> find;
+        if (index < (size / 2)) {
+            find = head;
+            for (int i = 0; i < index; i++) {
+                find = find.next;
+            }
+        } else {
+            find = tail;
+            for (int i = size - 1; i > index; i--) {
+                find = find.prev;
+            }
+        }
+        return find;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> t = tail;
```

Comment:
do we need this variable t? can we remove it?

Reply:
Maybe we could removed it if there was a different implementation. But in this code on 42 line we have this "tail = newNode;" and if we use node tail on our verification "if(tail == null ) , on this case it's will be always false and at line 46 the same (I don't know if I should answer here, sorry if not :) )